server/task_check_executor_lgtm.go

```diff
@@ -0,0 +1,143 @@
+package server
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/common"
+	"github.com/bytebase/bytebase/store"
+)
+
+// NewTaskCheckLGTMExecutor creates a task check LGTM executor.
+func NewTaskCheckLGTMExecutor() TaskCheckExecutor {
+	return &TaskCheckLGTMExecutor{}
+}
+
+// TaskCheckLGTMExecutor is the task check LGTM executor. It checks if "LGTM" comments are present.
+type TaskCheckLGTMExecutor struct {
+}
+
+// Run will run the task check LGTM executor once.
+func (*TaskCheckLGTMExecutor) Run(ctx context.Context, server *Server, taskCheckRun *api.TaskCheckRun) (result []api.TaskCheckResult, err error) {
+	task, err := server.store.GetTaskByID(ctx, taskCheckRun.TaskID)
+	if err != nil {
+		return []api.TaskCheckResult{}, common.Wrap(err, common.Internal)
+	}
+	if task == nil {
+		return []api.TaskCheckResult{
+			{
+				Status:    api.TaskCheckStatusError,
+				Namespace: api.BBNamespace,
+				Code:      common.Internal.Int(),
+				Title:     fmt.Sprintf("Failed to find task %v", taskCheckRun.TaskID),
+				Content:   err.Error(),
+			},
+		}, nil
+	}
+
+	issue, err := server.store.GetIssueByPipelineID(ctx, task.PipelineID)
+	if err != nil {
+		return []api.TaskCheckResult{}, common.Wrap(err, common.Internal)
+	}
+	if issue == nil {
+		return []api.TaskCheckResult{
+			{
+				Status:    api.TaskCheckStatusError,
+				Namespace: api.BBNamespace,
+				Code:      common.Internal.Int(),
+				Title:     fmt.Sprintf("Failed to find issue by pipelineID %v", task.PipelineID),
+				Content:   err.Error(),
+			},
+		}, nil
+	}
+
+	if issue.Project.LGTMCheckSetting.Value == api.LGTMValueDisabled {
+		return []api.TaskCheckResult{
+			{
+				Status:    api.TaskCheckStatusSuccess,
+				Namespace: api.BBNamespace,
+				Code:      common.Ok.Int(),
+				Title:     "Skip check",
+				Content:   "This check is disabled.",
+			},
+		}, nil
+	}
+
+	activityType := string(api.ActivityIssueCommentCreate)
+	activityList, err := server.store.FindActivity(ctx, &api.ActivityFind{
+		TypePrefix:  &activityType,
+		ContainerID: &issue.ID,
+	})
+	if err != nil {
+		return []api.TaskCheckResult{}, common.Wrap(err, common.Internal)
+	}
+
+	ok, err := checkLGTMcomments(ctx, server.store, activityList, issue)
+	if err != nil {
+		return []api.TaskCheckResult{}, errors.Wrap(err, "failed to check LGTM comments")
+	}
+	if ok {
+		return []api.TaskCheckResult{
+			{
+				Status:    api.TaskCheckStatusSuccess,
+				Namespace: api.BBNamespace,
+				Code:      common.Ok.Int(),
+				Title:     "OK",
+				Content:   "Valid LGTM found",
```

Comment:
nit: "LGTM found"
There seems no invalid LGTM?

Reply:
Given that the setting is `PROJECT_OWNER`, an LGTM by a project member is considered invalid, IMO.

Reply:
SG.